src/prefect/artifacts.py

```diff
@@ -0,0 +1,160 @@
+from typing import Optional
+
+from prefect import context, Client
+
+
+def running_with_backend() -> bool:
+    """
+    Determine if running in context of a backend. This is always true when running
+    using the `CloudTaskRunner`.
+
+    Returns:
+        - bool: if `running_with_backend` is set in context
+    """
+    return bool(context.get("running_with_backend"))
+
+
+def create_link(link: str) -> Optional[str]:
+    """
+    Create a link artifact
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - link (str): the link to post
+
+    Returns:
+        - str: the task run artifact ID
+    """
+    if not running_with_backend():
+        return None
+
+    client = Client()
+    return client.create_task_run_artifact(
+        task_run_id=context.get("task_run_id"), kind="link", data={"link": link}
+    )
+
+
+def update_link(task_run_artifact_id: str, link: str) -> bool:
+    """
+    Update an existing link artifact. This function will replace the current link
+    artifact with the new link provided.
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - task_run_artifact_id (str): the ID of an existing task run artifact
+        - link (str): the new link to update the artifact with
+
+    Returns:
+        - bool: whether or not the request was successful
+    """
+    if not running_with_backend():
+        return False
+
+    if task_run_artifact_id is None:
+        raise ValueError("The ID of an existing task run artifact must be provided.")
+
+    client = Client()
+    return client.update_task_run_artifact(
+        task_run_artifact_id=task_run_artifact_id, data={"link": link}
+    )
+
+
+def delete_link(task_run_artifact_id: str) -> bool:
+    """
+    Delete an existing link artifact
+
+    Note: The functionality here is experimental, and may change between
+    versions without notice. Use at your own risk.
+
+    Args:
+        - task_run_artifact_id (str): the ID of an existing task run artifact
+
+    Returns:
+        - bool: whether or not the request was successful
```

Comment:
Under what circumstances would the request be unsuccessful. Api failures will error, I assume this just means "did the id exist before"?

Reply:
The API returns a boolean whether or not the row in the DB was updated

```python
return bool(result.affected_rows)
```

I can update the docstring to make this clearer

Reply:
If `task_run_artifact_id` doesn't exist, what happens (error? or returns false?).

I personally don't see a user for a return value here, dropping it from `update` and `delete` makes the most sense to me.

Reply:
If it doesn't exist then an error is raised